### ISSUE TITLE
Codename - Outbreak

### ISSUE THREAD

**Issue by rickomax** [elishacloud/dxwrapper#226]
![image](https://github.com/elishacloud/dxwrapper/assets/12863984/099d6980-2817-4cbd-8958-f29cdccf52de)
Codename - Outbreak textures are recognizing black color as transparent when it shouldn't.

**Comment by elishacloud** [comment]
I don't have the ability to test this game right now.  However, last time I tested this game it required CreateExecuteBuffer, which I have not implemented yet.  Can you attach your log file for this?

**Comment by rickomax** [comment]
Sure!

```
11084 22:50:10.787 Starting DxWrapper v1.0.6542.21
11084 22:50:10.787 Running from: E:\GOG Games\Codename - Outbreak\dxwrapper.dll
11084 22:50:10.787 Reading config file: E:\GOG Games\Codename - Outbreak\dxwrapper.ini
11084 22:50:10.788 Gigabyte Technology Co., Ltd. A520M DS3H Default string (Desktop)
11084 22:50:10.788 Gigabyte Technology Co., Ltd. A520M DS3H x.x (Desktop)
11084 22:50:10.788 NVIDIA GeForce GTX 1060 6GB
11084 22:50:10.788 Windows 10 Enterprise 64-bit (10.0.19041)
11084 22:50:10.788 "OutBreak.exe" (PID:13788)
11084 22:50:10.792 Environment variable __COMPAT_LAYER = "DisableThemes HighDpiAware"
11084 22:50:10.793 Disabling High DPI Scaling...
11084 22:50:10.793 Loaded library: user32.dll
11084 22:50:10.793 Loaded library: shcore.dll
11084 22:50:10.793 DxWrapper loaded!
11084 22:50:37.024 Quiting DxWrapper
11084 22:50:37.024 Unloading libraries...
11084 22:50:37.024 Reseting screen resolution
11084 22:50:37.033 Reseting font smoothing
11084 22:50:37.035 DxWrapper terminated!
```

**Comment by rickomax** [comment]
After getting some info about the original engine, looks like it uses DDRAW 5.
Trying to use dxHwnd to run it, opens the menu screen, then produces a crash:

**Comment by eddmanx** [comment]
It's a D3D7 game, not just DDraw. dxwrapper doesn't yet properly support D3D. You have to use a different wrapper like dgvoodoo2.

**Comment by elishacloud** [comment]
> Trying to use dxHwnd to run it, opens the menu screen, then produces a crash:

DxWnd is a different tool.  You may want to ask the author of that tool.  I hope to have support for this game at some point, but I don't at this time.

**Comment by Mitradis** [comment]
this game use D3DImm.dll and DDraw.dll for wrapped.

**Comment by elishacloud** [comment]
dxwrapper includes functions for both D3DImm.dll and DDraw.dll.  However, there is still limited support for 3D games.  This one is not yet supported with the `dd7to9` function yet.

**Comment by Mitradis** [comment]
need restore this in config after config close.
TextureFormat1=32 0 0 0 64 0 0 0 0 0 0 0 16 0 0 0 0 248 0 0 224 7 0 0 31 0 0 0 0 0 0 0   
TextureFormat2=32 0 0 0 65 0 0 0 0 0 0 0 32 0 0 0 0 0 255 0 0 255 0 0 255 0 0 0 0 0 0 255

**Comment by elishacloud** [comment]
Yes, I have seen this type of issue with other games, where the config gets messed up.  I am not sure if it is caused by something in dxwrapper or a difference in how a newer OS responds to the API calls the older game makes.  I suspect the latter.

**Comment by elishacloud** [comment]
Can you try with the latest release build and send me the log files if it does not work?

**Comment by elishacloud** [comment]
I just tested this game and it uses Direct3D interface version 1, which is not yet supported by dxwrapper.

**Comment by elishacloud** [comment]
I added partial support for Direct3D interface version 1.  I tested with Codename - Outbreak and the game is playable.

Here is the updated version:  [dxwrapper.zip](https://github.com/user-attachments/files/18433986/dxwrapper.zip)

**Comment by elishacloud** [comment]
This latest build should address all the issues with this game:  [dxwrapper.zip](https://github.com/user-attachments/files/18527930/dxwrapper.zip)

**Comment by elishacloud** [comment]
Closing this issue.  It is fixed in the latest release.  Just make sure to enable the `DdrawIntroVideoFix` option.
